### PR TITLE
ref_task_4797

### PR DESCRIPTION
**Pull request by thanh01684265170**
Link: https://edu-redmine.framgia.vn/issues/4797

### CODE REVIEW

**Comment by thanh01684265170** [comment]
ready

File: app/src/main/java/com/framgia/toeic/data/model/Grammar.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package com.framgia.toeic.data.model;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+public class Grammar extends Question implements Parcelable {
+    private int mId;
+    private boolean mCheck;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/data/model/Grammar.java]: rename to mIsSelected

File: app/src/main/java/com/framgia/toeic/data/model/Grammar.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Override
+    public void writeToParcel(Parcel parcel, int i) {
+        parcel.writeInt(mId);
+        parcel.writeByte((byte) (mCheck ? 1 : 0));
+    }
+
+    public static class GrammarBuilder extends QuestionBuilder {
+        private int mId;
+        private boolean mCheck;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/data/model/Grammar.java]: the same

File: app/src/main/java/com/framgia/toeic/data/source/local/GrammarLessonDatabaseHelper.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                null, null, null, null, null);
+        cursorLesson.moveToFirst();
+        do {
+            GrammarLesson grammarLesson;
+            grammarLesson = new GrammarLesson(cursorLesson.getInt(
+                    cursorLesson.getColumnIndex(COLUMN_ID_LESSON)),
+                    cursorLesson.getString(cursorLesson.getColumnIndex(COLUMN_NAME)));
+            grammarLessons.add(grammarLesson);
+        } while (cursorLesson.moveToNext());
+        Log.d("Thanh", "Size: " + grammarLessons.size());
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/data/source/local/GrammarLessonDatabaseHelper.java]: remove log

File: app/src/main/java/com/framgia/toeic/data/source/local/GrammarLessonLocalDataSource.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.framgia.toeic.data.source.local;
+
+import com.framgia.toeic.data.GrammarLessonDataSource;
+import com.framgia.toeic.data.model.GrammarLesson;
+import com.framgia.toeic.data.source.Callback;
+
+import java.util.List;
+
+public class GrammarLessonLocalDataSource implements GrammarLessonDataSource.Local {
+    GrammarLessonDatabaseHelper mDatabaseHelper;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/data/source/local/GrammarLessonLocalDataSource.java]:
```suggestion
    private GrammarLessonDatabaseHelper mDatabaseHelper;
```

**Comment by thanh01684265170** [comment]
ready

File: app/src/main/java/com/framgia/toeic/data/model/Grammar.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    public Grammar() {
+    }
+
+    public Grammar(Grammar.GrammarBuilder grammarBuilder) {
+        super(grammarBuilder);
+        mId = grammarBuilder.mId;
+        mIsSelected = grammarBuilder.mIsSelected;
+    }
+
+    protected Grammar(Parcel in) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/data/model/Grammar.java]: thiếu super rồi em

File: app/src/main/java/com/framgia/toeic/data/model/Grammar.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    };
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel parcel, int i) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/data/model/Grammar.java]: thiếu supper rồi

File: app/src/main/java/com/framgia/toeic/screen/grammar/GrammarPresenter.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    public void getGrammars() {
+        mRepository.getGrammarLessons(new Callback<List<GrammarLesson>>() {
+            @Override
+            public void onGetDataSuccess(List<GrammarLesson> grammarLessons) {
+                mView.showGrammars(grammarLessons);
+            }
+
+            @Override
+            public void onGetDataFail(Exception error) {
+
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/toeic/screen/grammar/GrammarPresenter.java]: handle error

**Comment by thanh01684265170** [comment]
ready

**Pull request merged**